### DISCUSSION TITLE
LVI Mitigation?

### DISCUSSION THREAD

**Discussion by xanderdunn** [occlum/occlum#842] [Q&A] [answered]
The [Load Value Injection attack vector](https://lviattack.eu) against Intel SGX successfully circumvents the security of SGX. Details from Intel [here](https://www.intel.com/content/www/us/en/developer/articles/technical/software-security-guidance/technical-documentation/load-value-injection.html). 

The [Intel SGX SDK](https://github.com/intel/linux-sgx) describes using their mitigated toolchain to mitigate the LVI vulnerability.
The [OpenEnclave SDK](https://github.com/openenclave/openenclave/blob/master/samples/helloworld/README.md) has a section dedicated to building an enclave with LVI mitigation.

Searching this repo for "LVI", I didn't find any results. Does occlum support LVI mitigated SGX enclaves?

**Comment by guzongmin** [comment, marked as answer]
LVI mitigation is used for those Enclaves running on old CPUs, so by default Occlum does not open the LVI mitigation flag.
But you are able to open the mitigation by changing the Occlum build script. The Occlum docker file is https://github.com/occlum/occlum/blob/master/tools/docker/Dockerfile.ubuntu20.04. Since Occlum depends on the Intel SGX SDK, user is able to reference the Intel SGX SDK LVI mitigation tool chain to open it.
BTW: If you are using Intel SGX2 with the latest BIOS, you may not need the LVI mitigation.
